core/trino-main/src/main/java/io/trino/util/DateTimeUtils.java

```diff
@@ -67,9 +70,78 @@ public static int parseDate(String value)
         // - the required format is 'YYYY-MM-DD'
         // - all components should be unsigned numbers
         // https://github.com/trinodb/trino/issues/10677
+        OptionalInt days = parseIfIso8861DateFormat(value);
+        if (days.isPresent()) {
+            return days.getAsInt();
+        }
         return toIntExact(TimeUnit.MILLISECONDS.toDays(DATE_FORMATTER.parseMillis(value)));
     }
 
+    /**
+     * Check if the date is in format yyyy-MM-dd, if yes, return the epoch days; otherwise
+     * return OptionalInt.empty(), indicating the parse failed, and should fall back to generic
+     * parsing. Mixing the format checking and parsing in one pass for better performance.
+     * @param value date string to be parsed
+     * @return the number of days since 1970-01-01
+     */
+    @VisibleForTesting
+    static OptionalInt parseIfIso8861DateFormat(String value)
+    {
+        if (value.length() != 10) {
+            return OptionalInt.empty();
+        }
+        // year
+        OptionalInt year = parseIntSimple(value, 0, 4);
+        if (year.isEmpty()) {
+            return OptionalInt.empty();
+        }
+
+        if (value.charAt(4) != '-') {
+            return OptionalInt.empty();
+        }
+
+        // month
+        OptionalInt month = parseIntSimple(value, 5, 2);
+        if (month.isEmpty()) {
+            return OptionalInt.empty();
+        }
+        if (value.charAt(7) != '-') {
+            return OptionalInt.empty();
+        }
+
+        // day
+        OptionalInt day = parseIntSimple(value, 8, 2);
+        if (day.isEmpty()) {
+            return OptionalInt.empty();
+        }
+        LocalDate ld = LocalDate.of(year.getAsInt(), month.getAsInt(), day.getAsInt());
+        return OptionalInt.of(toIntExact(ld.toEpochDay()));
+    }
+    /**
+     * parse positive integer with radix 10, with no null or length check.
+     * @param str The string to be parsed
+     * @param start start position
+     * @param len number of chars to parse, and must >= 1
+     * @return the int, or OptionalInt.empty() if any non digit found
+     */
+    private static OptionalInt parseIntSimple(String str, int start, int len) {
+        if (len <= 0) {
+            return OptionalInt.empty();
```

Comment:
"len number of chars to parse, and must >= 1" so throwing IAE would be appropriate